autogpt_platform/backend/backend/server/v2/store/db.py

```diff
@@ -989,170 +1123,144 @@ async def get_admin_submissions(
-    """Get all submissions for a specific listing for admins.
-
-    Args:
-        listing_id: The ID of the store listing
-        page: Page number for pagination
-        page_size: Number of items per page
-
-    Returns:
-        StoreSubmissionsResponse with all versions of a specific listing
-    """
-    logger.debug(f"Getting submission history for listing {listing_id}")
-
-    try:
-        # Query StoreListingVersion directly since we need all versions for a listing
-        versions = await prisma.models.StoreListingVersion.prisma().find_many(
-            where={"storeListingId": listing_id},
-            order=[{"version": "desc"}],
-            skip=(page - 1) * page_size,
+        # Query listings with their versions
+        listings = await prisma.models.StoreListing.prisma().find_many(
+            where=where,
+            skip=skip,
             take=page_size,
-            include={"StoreListing": True},
+            include=include,
+            order=[{"createdAt": "desc"}],
         )
 
         # Get total count for pagination
-        total = await prisma.models.StoreListingVersion.prisma().count(
-            where={"storeListingId": listing_id}
-        )
+        total = await prisma.models.StoreListing.prisma().count(where=where)
         total_pages = (total + page_size - 1) // page_size
 
-        # Convert to StoreSubmission models
-        submissions = []
-        for version in versions:
-            if not version.StoreListing:
-                continue
-
-            submissions.append(
-                backend.server.v2.store.model.StoreSubmission(
-                    agent_id=version.agentId,
-                    agent_version=version.agentVersion,
-                    name=version.name,
-                    sub_heading=version.subHeading,
-                    slug=version.StoreListing.slug,
-                    description=version.description,
-                    image_urls=version.imageUrls or [],
-                    date_submitted=version.submittedAt or version.createdAt,
-                    status=version.submissionStatus,
-                    runs=0,  # We don't have this data here
-                    rating=0.0,  # We don't have this data here
-                    store_listing_version_id=version.id,
-                    reviewer_id=version.reviewerId,
-                    review_comments=version.reviewComments,
-                    internal_comments=version.internalComments,
-                    reviewed_at=version.reviewedAt,
-                    changes_summary=version.changesSummary,
+        # Convert to response models
+        listings_with_versions = []
+        for listing in listings:
+            versions: list[backend.server.v2.store.model.StoreSubmission] = []
+            # If we have versions, turn them into StoreSubmission models
+            if listing.Versions is not None:
+                for version in listing.Versions:
```

Comment:
nit:  you can always do
`for version in listing.Versions or []` to avoid code hadouken :D